selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/NavigationByKeyboardTest.java

```diff
@@ -90,32 +92,38 @@ public void setUp() throws Exception {
   public void navigationByKeyboard() throws Exception {
     projectExplorer.waitProjectExplorer();
     projectExplorer.waitItem(PROJECT_NAME);
-    projectExplorer.openItemByPath(PROJECT_NAME);
     notificationsPopupPanel.waitProgressPopupPanelClose();
     consoles.closeProcessesArea();
+    projectExplorer.selectItem(PROJECT_NAME);
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    projectExplorer.waitVisibleItem(PROJECT_NAME + "/src");
     projectExplorer.sendToItemDownArrowKey();
+    projectExplorer.waitItemIsSelected(PROJECT_NAME + "/src");
     projectExplorer.sendToItemEnterKey();
     projectExplorer.waitItemInVisibleArea("main");
     projectExplorer.waitItemInVisibleArea("test");
 
     projectExplorer.sendToItemDownArrowKey();
+    projectExplorer.waitItemIsSelected(PROJECT_NAME + "/src/main");
     projectExplorer.sendToItemRightArrowKey();
     projectExplorer.waitItemInVisibleArea("java");
     projectExplorer.waitItemInVisibleArea("webapp");
     projectExplorer.sendToItemDownArrowKey();
+    projectExplorer.waitItemIsSelected(PROJECT_NAME + "/src/main/java");
     projectExplorer.sendToItemEnterKey();
     projectExplorer.waitItemInVisibleArea(nameSecondModule);
     projectExplorer.waitItemInVisibleArea("com.example");
 
-    projectExplorer.selectItem(
-        PROJECT_NAME + "/src/main/java/" + nameSecondModule.replace('.', '/'));
+    projectExplorer.selectItem(PATH_TO_SECOND_MODULE);
+    projectExplorer.waitItemIsSelected(PATH_TO_SECOND_MODULE);
     projectExplorer.sendToItemRightArrowKey();
     checkItemsOfTheSecondModuleIsVisible();
     projectExplorer.sendToItemLeftArrowKey();
     checkItemsOfTheSecondModuleIsNotVisible();
     projectExplorer.sendToItemEnterKey();
     checkItemsOfTheSecondModuleIsVisible();
     projectExplorer.sendToItemDownArrowKey();
+    projectExplorer.waitItemIsSelected(PATH_TO_SECOND_MODULE + "/AppController.java");
```

Comment:
Please, divide separate steps by empty lines and comment them to improve readability of test.